source/Ship.cpp

```diff
@@ -1598,9 +1612,18 @@ void Ship::DoGeneration()
 		double coolingEfficiency = CoolingEfficiency();
 		energy += attributes.Get("energy generation") - attributes.Get("energy consumption");
 		energy -= ionization;
+		fuel += attributes.Get("fuel generation");
 		heat += attributes.Get("heat generation");
 		heat -= coolingEfficiency * attributes.Get("cooling");
 		
+		// Apply fuel to energy and heat conversion
```

Comment:
`// Convert fuel into energy and heat only when the required amount is available.`